compose/config/config.py

```diff
@@ -725,6 +726,20 @@ def split_label(label):
         return label, ''
 
 
+def merge_ulimits(base, override):
+    ulimits = parse_ulimits(base)
+    ulimits.update(parse_ulimits(override))
+    return ulimits
```

Comment:
I swear that we'd already fixed this a couple times, but apparently not, and now I can't find any open PRs with the fix.

I think all of these merge functions (`merge_labels`, `merge_environment`) should be just a single `merge_mapping` function that takes a function to parse the base and override.

Would you mind making that change instead of adding a new one?